src/CodeGen_Xtensa_vectors.template.cpp

```diff
@@ -2205,13 +2218,23 @@ HALIDE_ALWAYS_INLINE native_vector_i32_x4 convert<native_vector_i32_x4, native_v
                                 IVP_CVT32S2NX24HL(src), IVP_CVT32S2NX24HH(src));
 }
 
+template<>
+HALIDE_ALWAYS_INLINE native_vector_i32_x4 convert<native_vector_i32_x4, native_vector_i8>(const native_vector_i8 &src) {
+    xb_vec2Nx24 wide = src * native_vector_i8(1);
+    return native_vector_i32_x4(
+        native_vector_i32_x4::from_native_vector,
+        IVP_CVT32S2NX24LL(wide),
+        IVP_CVT32S2NX24LH(wide),
+        IVP_CVT32S2NX24HL(wide),
+        IVP_CVT32S2NX24HH(wide));
+}
+
 template<>
 HALIDE_ALWAYS_INLINE native_vector_i32_x2 convert<native_vector_i32_x2, native_vector_i16>(const native_vector_i16 &src) {
+    xb_vec2Nx24 wide = IVP_CVT24S2NX16(0, src);
     return native_vector_i32_x2(native_vector_i32_x2::from_native_vector,
```

Comment:
Oh, yeah, that was incorrect before for sure, because it needs to have sign extension. In my experiment a while ago using a shuffle/select was more efficient than going through the wide vector. Maybe we can use the old approach (which is currently incorrect), but add a sign extension as in halide_xtensa_convert_i16_low_i32/halide_xtensa_convert_i16_high_i32?